src/Disks/VolumeJBOD.h

```diff
@@ -38,11 +55,18 @@ class VolumeJBOD : public IVolume
     /// Returns valid reservation or nullptr if there is no space left on any disk.
     ReservationPtr reserve(UInt64 bytes) override;
 
+    bool areMergesAllowed() const override;
+
+    void setAllowMergesFromQuery(bool allow) override;
+
+    /// True if parts on this volume participate in merges according to configuration.
+    bool are_merges_allowed_in_config = true;
+
+    /// True if parts on this volume participate in merges according to START/STOP MERGES ON VOLUME.
```

Comment:
I do not quite understand why we need to manage this flag separately from `are_merges_allowed_in_config`. Could you please comment on this?

Reply:
This setting comes from two different sources, configuration (which can be reloaded) and user will (START / STOP). These two flags simplify conflict management of these two sources.